bd-logger/src/async_log_buffer.rs

```diff
@@ -690,10 +689,45 @@ impl<R: LogReplay + Send + 'static> AsyncLogBuffer<R> {
           if let Some(pre_config_log_buffer) = maybe_pre_config_buffer {
             self.maybe_replay_pre_config_buffer_logs(
                 pre_config_log_buffer,
-                &mut initial_logs
+                &mut vec![]
             ).await;
           }
         },
+        Ok((crash_monitor, session_id)) = &mut self.report_processor_rx => {
```

Comment:
I'm not 100% sure but I don't think a oneshot receiver is meant to be polled multiple times. I don't know if this will spin or do something else bad. I think you either need to put this into an option and then clear it when done, or fuse() it.